content/post/2019-10-15.Rmd

```diff
@@ -0,0 +1,388 @@
+
+This code populates our data frame using C++:
+
+```{r, warning = FALSE, message = FALSE}
+
+# Load in libaries
+library(Rcpp)
+library(tidyverse)
+
+# Here comes the C++ code
+cppFunction('DataFrame createTrajectory(int n, double x0, double y0,
+            double a, double b, double c, double d) {
+            // create the columns
+            NumericVector x(n);
+            NumericVector y(n);
+            x[0]=x0;
+            y[0]=y0;
+            for(int i = 1; i < n; ++i) {
+            x[i] = sin(a*y[i-1])+c*cos(a*x[i-1]);
+            y[i] = sin(b*x[i-1])+d*cos(b*y[i-1]);
+            }
+            // return a new data frame
+            return DataFrame::create(_["x"]= x, _["y"]= y);
+            }
+            ')
+
+a <- -1.25
+b <- -1.25
+c <- -1.82
+d <- -1.91
+
+# Call the function starting at (0,0) to store 10M jumps
+df <- createTrajectory(10000000, 0, 0, a, b, c, d)
+
+head(df)
+```
+
+Note that, apart of `Rcpp`,  we also loaded the tidyverse package. This is becaus we will use some of its features, as `dplyr` or `ggplot2`, later. 
+
+The function `createTrajectory` is written in `C++` and stores the itinerary of a initial point. You can try it to see that It's extremely fast. In the previous example, I populated on the fly our data frame with 10 milion points starting at `(0,0)` in less than 5 seconds in a conventional laptop.
+
+## Rendering the plot with ggplot
+
+Once we have build our data frame, it's time to represent it with `ggplot`. 
+
+Making graphics is a necessary skill and for this `ggplot` is an amazing tool, which by
+the way, I encourage you to learn. Usually, a good visualization is the best way to show your results. R is an extraordinary tool for visualizing data and one of the most important libraries to do it is `ggplot2`: it is powerful, flexible and is updated to include functionalities regularly. Although its syntax may seem a bit strange at first, here are four important concepts from my point of view to help you out with understanding it:
+
++ It’s truly important to have clear the type of graph you want to plot: for each one there’s a different *geometry*: `geom_point` to represent points, `geom_line` to represent time series, `geom_bar` to do bar chars ... Have a wonder though `ggplot2` package, there are many other types.
++ Each of those geometries needs to define what data is to be used and how. For example, as you will se in our graph, we placed on the x-axis the `x` column of our data frame `df` and `y` on the y-axis, so we did `aes(x = x, y = y)`.
++ You can also combine geometries. We didn’t do that for our example but you may want, for example, to combine bars and lines on a same plot. To do it you can just use `+` operator. This operator is very important: it doesn’t only combine geometries, it’s also used to add features to the plot as we did to define its aspect ratio (with `coord_equal`) or its appearance (with `theme_void`).
```

Comment:
1/ Creo que es más correcto: to combine bars and lines on THE same plot.

Usar The en vez de A.